samples/sample_common/src/sample_utils.cpp

```diff
@@ -180,16 +180,27 @@ mfxStatus CSmplYUVReader::LoadNextFrame(mfxFrameSurface1* pSurface)
 
     mfxU32 nBytesPerPixel = (pInfo.FourCC == MFX_FOURCC_P010 || pInfo.FourCC == MFX_FOURCC_P210 ) ? 2 : 1;
 
-    if (MFX_FOURCC_YUY2 == pInfo.FourCC || MFX_FOURCC_RGB4 == pInfo.FourCC || MFX_FOURCC_BGR4 == pInfo.FourCC || pInfo.FourCC == MFX_FOURCC_Y210 || pInfo.FourCC == MFX_FOURCC_Y410)
+    if (   MFX_FOURCC_YUY2 == pInfo.FourCC
+        || MFX_FOURCC_RGB4 == pInfo.FourCC
+        || MFX_FOURCC_BGR4 == pInfo.FourCC
+        || MFX_FOURCC_AYUV == pInfo.FourCC
+        || MFX_FOURCC_A2RGB10 == pInfo.FourCC
+#if (MFX_VERSION >= 1027)
+        || MFX_FOURCC_Y210 == pInfo.FourCC
+        || MFX_FOURCC_Y410 == pInfo.FourCC
+#endif
+    )
     {
         //Packed format: Luminance and chrominance are on the same plane
         switch (m_ColorFormat)
         {
+        case MFX_FOURCC_A2RGB10:
```

Comment:
In this way we mix pData.R and pData.Y. I don't think community will like this. Please, separate out handling of RGB and 4:2:0 color formats even if the code looks the same subject to R/Y difference.

Reply:
That way of storing pointers already implemented in allocator, using B pointer for all cases. That code is already in use, so I don't want to change it

Reply:
as we discussed, this community request is not related to this particular issue. This mix of pointers happened due to  bad code in allocators. I'll apply fix for both allocators and these functions.